src/odata-to-abstract-sql.ts

```diff
@@ -1080,6 +1080,30 @@ export class OData2AbstractSQL {
 			// It's the result of a lambda
 			return prop;
 		} else {
+			// Natively created timestamps in postgres have a microseconds precision
+			// js has only milliseconds precision, thus retrieved timestamps may fail on eq, ne, gt, lt comparisons
+			// furthermore retrieved timestamps should be truncated on database and not on the db abstraction layer
+			// thus SBVR Date Time fields have to be truncated to millisecond precision
+			// getting schema dataType from field mapping
+			let fieldDefinition;
+			const mapping = this.ResourceMapping(prop.resource);
+			if (mapping[prop.name]) {
+				fieldDefinition = _.find(prop.resource.fields, {
+					fieldName: mapping[prop.name][1],
+				});
```

Comment:
Can you use the native version in preference to lodash, it works better with typescript and should also be more performant:
```suggestion
				fieldDefinition = resource.fields.find(
					(f) => f.fieldName === mapping[prop.name][1],
				);
```